lib/irb/ruby-lex.rb

```diff
@@ -269,18 +275,15 @@ def each_top_level_statement
     end
   end
 
-  def lex
+  def lex(context: nil)
```

Comment:
Is this and the below context passing necessary? It looks like `lex` is never called with `context` argument?

Reply:
This context is mainly used to test RubyLex's nesting_level and code_block_open calculation in test_ruby_lex.rb

I fixed to call lex with context in L248.

I also changed the optional keyword parameter `context:` to normal required parameter.
- Not to forget calling method without context
- Method in this library that receives only context is using normal parameter, not keyword parameter. (ex: `set_auto_indent(context)`, `suspend_context(context)`)

Reply:
> Is it necessary?

Yes.
Since RubyLex#lex just returns @input.call in mulitline mode, this code is needed only when USE_MULTILINE is false.
Example when nil is passed to context here:
```
% irb --nomultiline
irb(main):001:0> a = 8
=> 8
irb(main):002:0> a /2 #=> should be 4
irb(main):003:0/ 
irb(main):004:0/ 
```